Misc fixes

simondeziel opened the pull request.

simondeziel committed:
github: use Go 1.21.x for lxd-snapcraft

Signed-off-by: Simon Deziel <simon.deziel@canonical.com>

simondeziel committed:
snapcraft: drop build-attributes

Signed-off-by: Simon Deziel <simon.deziel@canonical.com>

simondeziel committed:
snapcraft: s/SNAPCRAFT_/CRAFT_/

Signed-off-by: Simon Deziel <simon.deziel@canonical.com>

simondeziel commented:
@masnax it seems the raft part is build with liblz4 support but the corresponding .so doesn't make it to the resulting snap as it's not primed:

```
root@c1:/snap/microcloud/current# ll lib/
total 2309
lrwxrwxrwx 1 root root      18 Feb  2 18:14 libdqlite.so -> libdqlite.so.0.0.1*
lrwxrwxrwx 1 root root      18 Feb  2 18:14 libdqlite.so.0 -> libdqlite.so.0.0.1*
-rwxr-xr-x 1 root root 1251240 Feb  2 18:14 libdqlite.so.0.0.1*
lrwxrwxrwx 1 root root      16 Feb  2 18:14 libraft.so -> libraft.so.3.0.0*
lrwxrwxrwx 1 root root      16 Feb  2 18:14 libraft.so.3 -> libraft.so.3.0.0*
-rwxr-xr-x 1 root root 1113976 Feb  2 18:14 libraft.so.3.0.0*
drwxr-xr-x 2 root root       0 Feb  2 18:14 x86_64-linux-gnu/
root@c1:/snap/microcloud/current# ll lib/x86_64-linux-gnu/
total 194
lrwxrwxrwx 1 root root     14 Jan 15  2022 libuv.so.1 -> libuv.so.1.0.0
-rw-r--r-- 1 root root 198744 Jan 15  2022 libuv.so.1.0.0
```

Should I prime it? Or maybe disable lz4 support?

masnax commented:
Well MicroCloud's database is very basic (it's only a record of cluster members that it's sure are part of the MicroCloud in case the user messes with the other services) so I think disabling lz4 should be fine. 

Would we be able to turn it on if for some reason MicroCloud's database gets really complex and we want the extra compression?

There was also a performance benefit to turning it off, right?

simondeziel commented:
> Well MicroCloud's database is very basic (it's only a record of cluster members that it's sure are part of the MicroCloud in case the user messes with the other services) so I think disabling lz4 should be fine.

Agreed, done.

> There was also a performance benefit to turning it off, right?

So I've heard.

simondeziel committed:
snapcraft: make it clear that lz4 lib isn't primed

snapcraft deduplication is not applied to files from packages listed in
stage-packages.  To avoid duplicating lz4's .so, make it explicit that we don't
want to prime it and instead relying on the one provided in the base snap.

Signed-off-by: Simon Deziel <simon.deziel@canonical.com>

simondeziel commented:
@masnax I reverted to the default config for raft (so with lz4 support enabled) because it felt too risky for migrating existing DBs that would be lz4-compressed.

tomponline commented on snapcraft.yaml:
```diff
@@ -71,6 +71,7 @@ parts:
     prime:
       - lib/libraft*so*
       - lib/*/libuv.so*
+      - lib/*/liblz4.so*
```
@simondeziel why is this different compared to https://github.com/canonical/lxd-pkg-snap/pull/301/files#diff-e68b9afd67bb77dec8eda77796d3e0b784c7011203114996383eed6533be0cefR981 ?

simondeziel replied:
Thanks for catching this, it should have been the same. It is now.
